### PR TITLE
Refactory to Magento_Backend module class.

### PR DESCRIPTION
**Pull request by tiagosampaio**
Refactory of a class in Magento_Backend module.

### Description
Simple refactory to Magento_Backend module class.
- Turning protected properties into private ones.
- Using dependency injection rather than using Object Manager directly.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @tiagosampaio. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/Backend/Helper/Dashboard/Order.php [hunk 1]
```diff
@@ -18,62 +16,52 @@ class Order extends \Magento\Backend\Helper\Dashboard\AbstractDashboard
     /**
      * @var \Magento\Reports\Model\ResourceModel\Order\Collection
      */
-    protected $_orderCollection;
+    private $orderCollection;
```
Comment (ihor-sviziev) [inline app/code/Magento/Backend/Helper/Dashboard/Order.php]: This change is not backward compatible. Please revert it.
More details: https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/

File: app/code/Magento/Backend/Helper/Dashboard/Order.php [hunk 2]
```diff
@@ -18,62 +16,52 @@ class Order extends \Magento\Backend\Helper\Dashboard\AbstractDashboard
...
 
     /**
      * @var \Magento\Store\Model\StoreManagerInterface
      * @since 100.0.6
      */
-    protected $_storeManager;
+    private $storeManager;
```
Comment (ihor-sviziev) [inline app/code/Magento/Backend/Helper/Dashboard/Order.php]: This change is not backward compatible. Please revert it.
More details: https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development

File: app/code/Magento/Backend/Helper/Dashboard/Order.php [hunk 3]
```diff
@@ -18,62 +16,52 @@ class Order extends \Magento\Backend\Helper\Dashboard\AbstractDashboard
...
 
     /**
      * @param \Magento\Framework\App\Helper\Context $context
      * @param \Magento\Reports\Model\ResourceModel\Order\Collection $orderCollection
      */
     public function __construct(
         \Magento\Framework\App\Helper\Context $context,
-        \Magento\Reports\Model\ResourceModel\Order\Collection $orderCollection
+        \Magento\Reports\Model\ResourceModel\Order\Collection $orderCollection,
+        \Magento\Store\Model\StoreManagerInterface $storeManager
```
Comment (ihor-sviziev) [inline app/code/Magento/Backend/Helper/Dashboard/Order.php]: Please make new constructor argument optional and as retrieve it by fallback from object manager.
More details: https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by tiagosampaio** [comment]
@ihor-sviziev changes were reverted.

File: app/code/Magento/Backend/Helper/Dashboard/Order.php [hunk 4]
```diff
@@ -13,48 +13,41 @@
...
  * @since 100.0.2
  */
-class Order extends \Magento\Backend\Helper\Dashboard\AbstractDashboard
+class Order extends AbstractDashboard
 {
     /**
      * @var \Magento\Reports\Model\ResourceModel\Order\Collection
      */
-    protected $_orderCollection;
+    private $_orderCollection;
```
Comment (ihor-sviziev) [inline app/code/Magento/Backend/Helper/Dashboard/Order.php]: Please revert this change, it's not backward compatible

File: app/code/Magento/Backend/Helper/Dashboard/Order.php [hunk 5]
```diff
@@ -13,48 +13,41 @@
...
 
     /**
      * @var \Magento\Store\Model\StoreManagerInterface
      * @since 100.0.6
      */
-    protected $_storeManager;
+    private $_storeManager;
```
Comment (ihor-sviziev) [inline app/code/Magento/Backend/Helper/Dashboard/Order.php]: Please revert this change, it's not backward compatible

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
In general looks good, but there still backward incompatible changes

**Comment by ihor-sviziev** [comment]
I fixed these small issues myself. Let's wait for Travis results

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-2687](https://jira.corp.magento.com/browse/ENGCOM-2687) has been created to process this Pull Request

**Comment by tiagosampaio** [comment]
Thanks @ihor-sviziev.

**Comment by magento-engcom-team** [comment]
Hi @tiagosampaio. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.7. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**